bundles/org.eclipse.equinox.common.tests/src/org/eclipse/equinox/common/tests/PathTest.java

```diff
@@ -1042,4 +1042,32 @@ public void testUptoSegment() {
 		anyPath = new Path("//one/two/three/");
 		assertEquals("5.4", new Path("//"), anyPath.uptoSegment(0));
 	}
+
+	public void testToPath() {
+
+		//Case 1, absolute path with no trailing separator
+		IPath anyPath = new Path("/first/second/third");
+
+		assertNotNull("1.0", anyPath.toPath());
```

Comment:
IMHO those marker messages are not necessary. In case the assertion fails the line information provided by the stack trace is sufficient to identify the failing line. Such generic messages are often confusing. Only verbose messages add real value, but in most cases the line-info from the stack trace is sufficient with the code available.

Reply:
I share the same but I was following existing style in this file. I'd rather like to avoid introducing an inconsistency.

Reply:
> I'd rather like to avoid introducing an inconsistency.

In this case I prefer to have inconsistency. 

There is no one left in the project who understands the reasoning behind such "very special" assertion messages.

Reply:
Please don't write bad tests for "consistency" and choose e meaningful message (e.g "absoloute IPath must return a non null java Path)

Reply:
@HannesWell What's the protocol for test bundle version update? Should it be `3.15.600.qualifier` or `3.16.0.qualifier`?

Reply:
See https://wiki.eclipse.org/Version_Numbering#When_to_change_the_minor_segment

Reply:
Btw, here are related links I wish every committer should have in bookmarks:

- https://wiki.eclipse.org/Version_Numbering 
- https://wiki.eclipse.org/Evolving_Java-based_APIs
- https://wiki.eclipse.org/Evolving_Java-based_APIs_2
- https://wiki.eclipse.org/Evolving_Java-based_APIs_3

Reply:
You can add them for example here:
- https://github.com/eclipse-equinox/equinox/blob/master/CONTRIBUTING.md

Reply:
> > I'd rather like to avoid introducing an inconsistency.
> 
> In this case I prefer to have inconsistency.

Fully agree.



> You can add them for example here:
> 
>     * https://github.com/eclipse-equinox/equinox/blob/master/CONTRIBUTING.md

I think it would be better to add it to one umbrella repo, from which all others can reference so that we only have to maintain one file. Maybe in https://github.com/eclipse-platform/.github?

Reply:
> Btw, here are related links I wish every committer should have in bookmarks:
> 
> * https://wiki.eclipse.org/Version_Numbering
> * https://wiki.eclipse.org/Evolving_Java-based_APIs
> * https://wiki.eclipse.org/Evolving_Java-based_APIs_2
> * https://wiki.eclipse.org/Evolving_Java-based_APIs_3

Is this wiki going away soon?